ckanext/dalrrd_emc_dcpr/blueprints/error_report.py

```diff
@@ -0,0 +1,443 @@
+
+
+def _request_url_(params_nopage, requests_type, q=None, page=None):
+    params = list(params_nopage)
+    params.append(("page", page))
+    url = request.url_rule.rule
+
+    return url_with_params(url, params)
+
+
+def _get_error_reports_list(ckan_action: str, should_show_create_action: bool = False):
+    try:
+        error_reports = toolkit.get_action(ckan_action)(
+            context={
+                "user": toolkit.g.user,
+                "dictize_for_ui": True,
+            },
+            data_dict={},
+        )
+    except toolkit.NotAuthorized:
+        result = toolkit.abort(
+            403,
+            toolkit._("Not authorized to list error reports"),
+        )
+
+    else:
+        params_nopage = [
+            (k, v) for k, v in request.args.items(multi=True) if k != "page"
+        ]
+        params_nosort = [(k, v) for k, v in params_nopage]
+        pager_url = partial(_request_url_, params_nosort, None)
+        page = h.get_page_number(request.args)
+        extra_vars = {
+            "error_reports": error_reports,
+            "statuses": get_status_labels(),
+            "show_create_button": should_show_create_action,
+            "page": h.Page(
+                collection=error_reports,
+                items_per_page=20,
+                url=pager_url,
+                page=page,
+                item_count=len(error_reports),
+            ),
+        }
+        result = toolkit.render("error_report/list.html", extra_vars=extra_vars)
+    return result
+
+
+class ErrorReportCreateView(MethodView):
+    def get(self, data=None, errors=None, error_summary=None):
+        toolkit.check_access("error_report_create_auth", {"user": toolkit.g.user})
+        data_to_show = data or clean_dict(
+            dict_fns.unflatten(
+                tuplize_dict(parse_params(request.args, ignore_keys=CACHE_PARAMETERS))
+            )
+        )
+        packages = ckan.model.Session.query(ckan.model.Package).all()
+
+        metadata_records = [
+            {"value": record.id, "text": record.title} for record in packages
+        ]
+
+        selected_metadata_record = request.args.get("metadata_record", None)
+
+        extra_vars = {
+            "form_snippet": "error_report/snippets/report_form.html",
+            "enable_owner_fieldset": True,
+            "enable_nsif_fieldset": False,
+            "csi_reference_id": None,
+            "data": data_to_show,
+            "metadata_records": metadata_records,
+            "selected_record": selected_metadata_record,
+            "errors": errors or {},
+            "error_summary": error_summary or {},
+        }
+        return toolkit.render("error_report/edit.html", extra_vars=extra_vars)
+
+    def post(self):
+        try:
+            data_dict = clean_dict(
+                dict_fns.unflatten(tuplize_dict(parse_params(request.form)))
+            )
+        except dict_fns.DataError:
+            result = toolkit.abort(400, toolkit._("Integrity Error"))
```

Comment:
not necessarily a thing, but if we can show a more descriptive error message it will be great